In [337]:
%matplotlib inline
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

## Load the 3 data sets from Word Data Bank
***
### Country dataframe

In [338]:
country = pd.DataFrame.from_csv('data/Country.csv', index_col = None)
country.head()

CountryCode       ShortName       TableName  \
0         AFG     Afghanistan     Afghanistan   
1         ALB         Albania         Albania   
2         DZA         Algeria         Algeria   
3         ASM  American Samoa  American Samoa   
4         ADO         Andorra         Andorra   

                                  LongName Alpha2Code    CurrencyUnit  \
0             Islamic State of Afghanistan         AF  Afghan afghani   
1                      Republic of Albania         AL    Albanian lek   
2  People's Democratic Republic of Algeria         DZ  Algerian dinar   
3                           American Samoa         AS     U.S. dollar   
4                  Principality of Andorra         AD            Euro   

                                        SpecialNotes  \
0  Fiscal year end: March 20; reporting period fo...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                       Region           IncomeGroup Wb2Code  \
0                  South Asia            Low income      AF   
1       Europe & Central Asia   Upper middle income      AL   
2  Middle East & North Africa   Upper middle income      DZ   
3         East Asia & Pacific   Upper middle income      AS   
4       Europe & Central Asia  High income: nonOECD      AD   

             ...                 GovernmentAccountingConcept  \
0            ...             Consolidated central government   
1            ...                Budgetary central government   
2            ...                Budgetary central government   
3            ...                                         NaN   
4            ...                                         NaN   

               ImfDataDisseminationStandard  \
0  General Data Dissemination System (GDDS)   
1  General Data Dissemination System (GDDS)   
2  General Data Dissemination System (GDDS)   
3                                       NaN   
4                                       NaN   

                              LatestPopulationCensus  \
0                                               1979   
1                                               2011   
2                                               2008   
3                                               2010   
4  2011. Population data compiled from administra...   

                               LatestHouseholdSurvey  \
0  Multiple Indicator Cluster Survey (MICS), 2010/11   
1       Demographic and Health Survey (DHS), 2008/09   
2     Multiple Indicator Cluster Survey (MICS), 2012   
3                                                NaN   
4                                                NaN   

          SourceOfMostRecentIncomeAndExpenditureData  \
0            Integrated household survey (IHS), 2008   
1  Living Standards Measurement Study Survey (LSM...   
2            Integrated household survey (IHS), 1995   
3                                                NaN   
4                                                NaN   

  VitalRegistrationComplete LatestAgriculturalCensus LatestIndustrialData  \
0                       NaN                  2013/14                  NaN   
1                       Yes                     2012               2011.0   
2                       NaN                      NaN               2010.0   
3                       Yes                     2007                  NaN   
4                       Yes                      NaN                  NaN   

  LatestTradeData LatestWaterWithdrawalData  
0          2013.0                    2000.0  
1          2013.0                    2006.0  
2          2013.0                    2001.0  
3             NaN                       NaN  
4          2006.0                       NaN  

[5 rows x 31 columns]

In [339]:
country.shape

(247, 31)

The Country data frame has 31 descriptor columns for each of the 247 rows (countries of groups of countries). Let's get a sense for how many individual countries exist and how many aggregate groupings (e.g. world, regions, income groupings). 

In [340]:
# Aggregated geographical areas will not have a currency unit, so count how many rows have null currency
len(country[country.CurrencyUnit.isnull()])

33

In [341]:
# Also check for aggregated data rows by searching for 'aggregate' in the special notes column 
text = country.SpecialNotes.fillna('0')
len(text[text.str.contains('aggregate')])

33

### Load 'Series' dataframe (renamed as metrics)

In [342]:
metrics = pd.DataFrame.from_csv('data/Series.csv', index_col = None)
metrics.head()

SeriesCode                                              Topic  \
0        BN.KLT.DINV.CD  Economic Policy & Debt: Balance of payments: C...   
1  BX.KLT.DINV.WD.GD.ZS  Economic Policy & Debt: Balance of payments: C...   
2     BX.KLT.DINV.CD.WD  Economic Policy & Debt: Balance of payments: C...   
3     BM.KLT.DINV.GD.ZS  Economic Policy & Debt: Balance of payments: C...   
4        BN.TRF.KOGT.CD  Economic Policy & Debt: Balance of payments: C...   

                                       IndicatorName ShortDefinition  \
0  Foreign direct investment, net (BoP, current US$)             NaN   
1  Foreign direct investment, net inflows (% of GDP)             NaN   
2  Foreign direct investment, net inflows (BoP, c...             NaN   
3  Foreign direct investment, net outflows (% of ...             NaN   
4             Net capital account (BoP, current US$)             NaN   

                                      LongDefinition UnitOfMeasure  \
0  Foreign direct investment are the net inflows ...           NaN   
1  Foreign direct investment are the net inflows ...           NaN   
2  Foreign direct investment refers to direct inv...           NaN   
3  Foreign direct investment are the net inflows ...           NaN   
4  Net capital account records acquisitions and d...           NaN   

  Periodicity BasePeriod  OtherNotes AggregationMethod  \
0      Annual        NaN         NaN               NaN   
1      Annual        NaN         NaN  Weighted average   
2      Annual        NaN         NaN               Sum   
3      Annual        NaN         NaN  Weighted average   
4      Annual        NaN         NaN               NaN   

                            LimitationsAndExceptions NotesFromOriginalSource  \
0                                                NaN                     NaN   
1  FDI data do not give a complete picture of int...                     NaN   
2  FDI data do not give a complete picture of int...                     NaN   
3                                                NaN                     NaN   
4                                                NaN                     NaN   

                                     GeneralComments  \
0  Note: Data are based on the sixth edition of t...   
1  Note: Data starting from 2005 are based on the...   
2  Note: Data starting from 2005 are based on the...   
3  Note: Data are based on the sixth edition of t...   
4  Note: Data are based on the sixth edition of t...   

                                              Source  \
0  International Monetary Fund, Balance of Paymen...   
1  International Monetary Fund, International Fin...   
2  International Monetary Fund, Balance of Paymen...   
3  International Monetary Fund, International Fin...   
4  International Monetary Fund, Balance of Paymen...   

                    StatisticalConceptAndMethodology  \
0                                                NaN   
1  Data on equity flows are based on balance of p...   
2  Data on equity flows are based on balance of p...   
3                                                NaN   
4                                                NaN   

                                DevelopmentRelevance RelatedSourceLinks  \
0                                                NaN                NaN   
1  Private financial flows - equity and debt - ac...                NaN   
2  Private financial flows - equity and debt - ac...                NaN   
3                                                NaN                NaN   
4                                                NaN                NaN   

   OtherWebLinks  RelatedIndicators LicenseType  
0            NaN                NaN        Open  
1            NaN                NaN        Open  
2            NaN                NaN        Open  
3            NaN                NaN        Open  
4            NaN                NaN        Open

In [343]:
metrics.shape

(1345, 20)

One row per series/metric, columns provide defintions, units, periodicity sources, etc. Now let's get a better sense for the key categories (and sub-categories) of development indicators.

In [344]:
# Highlevel topic categories
splitTopics = metrics.Topic.str.partition(':')
splitTopics.groupby([0,2]).size()
# splitTopics[splitTopics[0].str.contains('Health')].groupby([0,2]).size()

0                          2                                                                          
Economic Policy & Debt      Balance of payments: Capital & financial account                              11
                            Balance of payments: Current account: Balances                                 4
                            Balance of payments: Current account: Goods, services & income                22
                            Balance of payments: Current account: Transfers                                7
                            Balance of payments: Reserves & other items                                    6
                            External debt: Amortization                                                   18
                            External debt: Arrears, reschedulings, etc.                                   22
                            External debt: Commitments                                                     5
                         

Interested in health/life expactancy as the target indicators, so let's consider all Health metrics available

In [345]:
# all topics related to health 
health =metrics[metrics.Topic.str.contains("Health")]
health.groupby(['Topic']).size()

Topic
Health: Disease prevention       18
Health: Health services          14
Health: Mortality                22
Health: Nutrition                24
Health: Population: Dynamics     11
Health: Population: Structure     5
Health: Reproductive health      14
Health: Risk factors             15
dtype: int64

In [346]:
metrics[metrics.Topic.str.contains('Mortality')].head()

SeriesCode              Topic  \
813  SH.DTH.COMM.ZS  Health: Mortality   
814  SH.DTH.INJR.ZS  Health: Mortality   
815  SH.DTH.NCOM.ZS  Health: Mortality   
816  SP.DTH.INFR.ZS  Health: Mortality   
817  SP.DTH.REPT.ZS  Health: Mortality   

                                         IndicatorName ShortDefinition  \
813  Cause of death, by communicable diseases and m...             NaN   
814             Cause of death, by injury (% of total)             NaN   
815  Cause of death, by non-communicable diseases (...             NaN   
816  Completeness of infant death reporting (% of r...             NaN   
817  Completeness of total death reporting (% of re...             NaN   

                                        LongDefinition UnitOfMeasure  \
813  Cause of death refers to the share of all deat...           NaN   
814  Cause of death refers to the share of all deat...           NaN   
815  Cause of death refers to the share of all deat...           NaN   
816  Completeness of infant death reporting is the ...           NaN   
817  Completeness of total death reporting is the n...           NaN   

    Periodicity BasePeriod  OtherNotes AggregationMethod  \
813      Annual        NaN         NaN  Weighted average   
814      Annual        NaN         NaN  Weighted average   
815      Annual        NaN         NaN  Weighted average   
816      Annual        NaN         NaN               NaN   
817      Annual        NaN         NaN  Weighted average   

                              LimitationsAndExceptions  \
813  The limited availability of data on health sta...   
814  The limited availability of data on health sta...   
815  The limited availability of data on health sta...   
816                                                NaN   
817                                                NaN   

    NotesFromOriginalSource GeneralComments  \
813                     NaN             NaN   
814                     NaN             NaN   
815                     NaN             NaN   
816                     NaN             NaN   
817                     NaN             NaN   

                                                Source  \
813  Derived based on the data from WHO's World Hea...   
814  Derived based on the data from WHO's World Hea...   
815  Derived based on the data from WHO's World Hea...   
816  The United Nations Statistics Division's Popul...   
817  The United Nations Statistics Division's Popul...   

                      StatisticalConceptAndMethodology DevelopmentRelevance  \
813  Data on cause of death are compiled by the WHO...                  NaN   
814  Data on cause of death are compiled by the WHO...                  NaN   
815  Data on cause of death are compiled by the WHO...                  NaN   
816                                                NaN                  NaN   
817                                                NaN                  NaN   

    RelatedSourceLinks  OtherWebLinks  RelatedIndicators LicenseType  
813                NaN            NaN                NaN        Open  
814                NaN            NaN                NaN        Open  
815                NaN            NaN                NaN        Open  
816                NaN            NaN                NaN        Open  
817                NaN            NaN                NaN        Open

As initial features that could affect health/life expectancy, let's consider all indicators related to GDP and povery.

In [347]:
# Topics specifically related to GDP per capita 
metrics.IndicatorName[metrics.IndicatorName.str.contains('GDP per capita')]

295                     GDP per capita growth (annual %)
309                        GDP per capita (constant LCU)
389                   GDP per capita (constant 2005 US$)
409                         GDP per capita (current LCU)
410                         GDP per capita (current US$)
493    GDP per capita, PPP (constant 2011 internation...
494        GDP per capita, PPP (current international $)
540    Government expenditure per student, primary (%...
541    Government expenditure per student, secondary ...
542    Government expenditure per student, tertiary (...
Name: IndicatorName, dtype: object

In [348]:
metrics[metrics.IndicatorName.str.contains('GDP per capita')].head()

SeriesCode                                              Topic  \
295  NY.GDP.PCAP.KD.ZG  Economic Policy & Debt: National accounts: Gro...   
309     NY.GDP.PCAP.KN  Economic Policy & Debt: National accounts: Loc...   
389     NY.GDP.PCAP.KD  Economic Policy & Debt: National accounts: US$...   
409     NY.GDP.PCAP.CN  Economic Policy & Debt: National accounts: US$...   
410     NY.GDP.PCAP.CD  Economic Policy & Debt: National accounts: US$...   

                          IndicatorName ShortDefinition  \
295    GDP per capita growth (annual %)             NaN   
309       GDP per capita (constant LCU)             NaN   
389  GDP per capita (constant 2005 US$)             NaN   
409        GDP per capita (current LCU)             NaN   
410        GDP per capita (current US$)             NaN   

                                        LongDefinition UnitOfMeasure  \
295  Annual percentage growth rate of GDP per capit...           NaN   
309  GDP per capita is gross domestic product divid...           NaN   
389  GDP per capita is gross domestic product divid...           NaN   
409  GDP per capita is gross domestic product divid...           NaN   
410  GDP per capita is gross domestic product divid...           NaN   

    Periodicity         BasePeriod  OtherNotes AggregationMethod  \
295      Annual                NaN         NaN  Weighted average   
309      Annual  varies by country         NaN               NaN   
389      Annual               2005         NaN  Weighted average   
409      Annual                NaN         NaN               NaN   
410      Annual                NaN         NaN  Weighted average   

    LimitationsAndExceptions NotesFromOriginalSource GeneralComments  \
295                      NaN                     NaN             NaN   
309                      NaN                     NaN             NaN   
389                      NaN                     NaN             NaN   
409                      NaN                     NaN             NaN   
410                      NaN                     NaN             NaN   

                                                Source  \
295  World Bank national accounts data, and OECD Na...   
309  World Bank national accounts data, and OECD Na...   
389  World Bank national accounts data, and OECD Na...   
409  World Bank national accounts data, and OECD Na...   
410  World Bank national accounts data, and OECD Na...   

                      StatisticalConceptAndMethodology DevelopmentRelevance  \
295  For more information, see the metadata for con...                  NaN   
309                                                NaN                  NaN   
389  For more information, see the metadata for con...                  NaN   
409                                                NaN                  NaN   
410  For more information, see the metadata for cur...                  NaN   

    RelatedSourceLinks  OtherWebLinks  RelatedIndicators LicenseType  
295                NaN            NaN                NaN        Open  
309                NaN            NaN                NaN        Open  
389                NaN            NaN                NaN        Open  
409                NaN            NaN                NaN        Open  
410                NaN            NaN                NaN        Open

In [349]:
# Also considering topics related to Poverty
economic =metrics[metrics.Topic.str.contains("Poverty")]
economic.groupby(['Topic']).size()

Topic
Poverty: Income distribution     8
Poverty: Poverty rates          10
Poverty: Shared prosperity       6
dtype: int64

### Load 'Indicators' dataframe (renamed as 'df')
***

In [350]:
df = pd.DataFrame.from_csv('data/indicators.csv', index_col=None)
df.head(5)

CountryName CountryCode                                      IndicatorName  \
0  Arab World         ARB  Adolescent fertility rate (births per 1,000 wo...   
1  Arab World         ARB  Age dependency ratio (% of working-age populat...   
2  Arab World         ARB  Age dependency ratio, old (% of working-age po...   
3  Arab World         ARB  Age dependency ratio, young (% of working-age ...   
4  Arab World         ARB        Arms exports (SIPRI trend indicator values)   

    IndicatorCode  Year         Value  
0     SP.ADO.TFRT  1960  1.335609e+02  
1     SP.POP.DPND  1960  8.779760e+01  
2  SP.POP.DPND.OL  1960  6.634579e+00  
3  SP.POP.DPND.YG  1960  8.102333e+01  
4  MS.MIL.XPRT.KD  1960  3.000000e+06

In [351]:
df.shape

(5656458, 6)

In [352]:
#checking availability of data and how missing values are handled
df[df.IndicatorName.str.contains('GDP per capita')].groupby(['CountryName', 'IndicatorName']).size().unstack()

IndicatorName           GDP per capita (constant 2005 US$)  \
CountryName                                                  
Afghanistan                                           13.0   
Albania                                               35.0   
Algeria                                               55.0   
American Samoa                                         NaN   
Andorra                                               44.0   
Angola                                                 1.0   
Antigua and Barbuda                                   38.0   
Arab World                                            40.0   
Argentina                                             55.0   
Armenia                                               25.0   
Aruba                                                 16.0   
Australia                                             55.0   
Austria                                               55.0   
Azerbaijan                                            25.0   
Bahamas, The                                          55.0   
Bahrain                                               35.0   
Bangladesh                                            55.0   
Barbados                                              35.0   
Belarus                                               25.0   
Belgium                                               55.0   
Belize                                                55.0   
Benin                                                 55.0   
Bermuda                                               54.0   
Bhutan                                                35.0   
Bolivia                                               55.0   
Bosnia and Herzegovina                                21.0   
Botswana                                              55.0   
Brazil                                                55.0   
Brunei Darussalam                                     41.0   
Bulgaria                                              35.0   
...                                                    ...   
Switzerland                                           35.0   
Syrian Arab Republic                                  48.0   
Tajikistan                                            30.0   
Tanzania                                              27.0   
Thailand                                              55.0   
Timor-Leste                                           16.0   
Togo                                                  55.0   
Tonga                                                 34.0   
Trinidad and Tobago                                   55.0   
Tunisia                                               50.0   
Turkey                                                55.0   
Turkmenistan                                          28.0   
Tuvalu                                                25.0   
Uganda                                                33.0   
Ukraine                                               28.0   
United Arab Emirates                                  40.0   
United Kingdom                                        55.0   
United States                                         55.0   
Upper middle income                                   55.0   
Uruguay                                               55.0   
Uzbekistan                                            28.0   
Vanuatu                                               36.0   
Venezuela, RB                                         55.0   
Vietnam                                               31.0   
Virgin Islands (U.S.)                                  NaN   
West Bank and Gaza                                    21.0   
World                                                 55.0   
Yemen, Rep.                                           24.0   
Zambia                                                55.0   
Zimbabwe                                              55.0   

IndicatorName           GDP per capita (constant LCU)  \
CountryName                         

In [353]:
#Confirming how missing values are handled; for Afganistan and 'GDP per capita (2005), the 13 values are consecutive year. 
#When data is not available, it is omitted rather than having a null value per that year
df[(df.IndicatorCode =='NY.GDP.PCAP.KD') & (df.CountryName == 'Afghanistan')].sort_values('Year')

CountryName CountryCode                       IndicatorName  \
3494844  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
3649897  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
3803733  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
3964959  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
4144830  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
4322452  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
4502763  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
4682308  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
4862365  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
5047724  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
5223950  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
5397193  Afghanistan         AFG  GDP per capita (constant 2005 US$)   
5546497  Afghanistan         AFG  GDP per capita (constant 2005 US$)   

          IndicatorCode  Year       Value  
3494844  NY.GDP.PCAP.KD  2002  239.699451  
3649897  NY.GDP.PCAP.KD  2003  248.156635  
3803733  NY.GDP.PCAP.KD  2004  240.184904  
3964959  NY.GDP.PCAP.KD  2005  257.175795  
4144830  NY.GDP.PCAP.KD  2006  263.012374  
4322452  NY.GDP.PCAP.KD  2007  291.128823  
4502763  NY.GDP.PCAP.KD  2008  294.238183  
4682308  NY.GDP.PCAP.KD  2009  347.208097  
4862365  NY.GDP.PCAP.KD  2010  366.324813  
5047724  NY.GDP.PCAP.KD  2011  377.292766  
5223950  NY.GDP.PCAP.KD  2012  418.426197  
5397193  NY.GDP.PCAP.KD  2013  413.335227  
5546497  NY.GDP.PCAP.KD  2014  406.248136

In [354]:
#Comparing the available Life Expectancy metrics, is there one that's much better populated? No.
df[df.IndicatorName.str.contains('Life expectancy')].groupby(['CountryName', 'IndicatorName']).size().unstack().head()

IndicatorName        Life expectancy at birth, female (years)  \
CountryName                                                     
Afghanistan                                                54   
Albania                                                    54   
Algeria                                                    54   
Angola                                                     54   
Antigua and Barbuda                                        54   

IndicatorName        Life expectancy at birth, male (years)  \
CountryName                                                   
Afghanistan                                              54   
Albania                                                  54   
Algeria                                                  54   
Angola                                                   54   
Antigua and Barbuda                                      54   

IndicatorName        Life expectancy at birth, total (years)  
CountryName                                                   
Afghanistan                                               54  
Albania                                                   54  
Algeria                                                   54  
Angola                                                    54  
Antigua and Barbuda                                       54

Selected metrics
- Target indicator of health / life expectancy, selected: SP.DYN.LE00.IN - Life expectancy at birth, total (years) which is populated for 54 years of data
- For initial features: 
Of per-capita-measures of GDP, the following is most-densely populated: NY.GDP.PCAP.CD	- GDP per capita (current US)

## Data wrangling & inferential statistics
*** 

1. Transform the data frame into a more workable state, using Pivot_table() 
2. Create a data frame for all countries with a multi-variate index (CountryName & year) and 4 initial data columns: GDP, the annual change in GDP (gdp_diff), life expectancy, and the annual change in life expectancy (le_diff)
3. Plot time series and a scatter plot to observe the relationship between GDP and life expectancy. Not surprisingly, both metrics have increased over the past 55 years. To be determined whether this is a causation or correlation (admittedly difficult to completely tease apart the numerous other confounding factors that influence life expectancy and GDP)
4. plot the correlation
5. plot some scatter plots between variables

In [439]:
df.IndicatorCode = df.IndicatorCode.astype(str)
df.IndicatorName = df.IndicatorName.astype(str)

# piovt the 2013 data so the features are column headers
pivot = df[df.Year==2013].pivot_table('Value','CountryName', 'IndicatorCode')

features = ['SP.DYN.LE00.IN', #Life expectancy at birth, total (years)
#             'SI.POV.GINI',    #GINI index (World Bank estimate) 
            'NY.GDP.PCAP.KD', #’GDP per capita (constant 2005 US\$)’
#             'SI.POV.NAHC', #‘Poverty headcount ratio at national poverty lines (% of population)’
#             'IQ.CPA.TRAN.XQ',
            'SH.XPD.PCAP', #‘Health expenditure per capita (current US\$)’
            'NY.ADJ.AEDU.GN.ZS']  #Adjusted savings: education expenditure (% of GNI)

df_features = pivot[features]
df_features.columns = df_features.columns.to_series()
flattened = pd.DataFrame(df_features.to_records())

df_ml = flattened.rename(index=str, columns={"SP.DYN.LE00.IN": "LifeExpect_total", 
#                                              'SI.POV.GINI': "GINI",
                                             "NY.GDP.PCAP.KD": "GDP_perCap", 
#                                              "SI.POV.NAHC": "Poverty_percent",
#                                              "IQ.CPA.TRAN.XQ": "Corruption",
                                             "SH.XPD.PCAP": "HealthExp_perCap",
                                             "NY.ADJ.AEDU.GN.ZS": "EduExp_percentGNI"})

# Drop countries who do not have a life expectancy value 
df_ml = df_ml[pd.notnull(df_ml.LifeExpect_total)]
df_ml = df_ml.dropna(axis =0, how='any')

In [440]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import Imputer

y = df_ml.LifeExpect_total.values.reshape(-1,1)
X = df_ml.drop('LifeExpect_total', axis = 1).drop('CountryName', axis = 1)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state=42) 

lr = LinearRegression(normalize=True)

lr.fit(X_train, y_train)
y_pred = lr.predict(X_test)
lr.score(X_test, y_test)

0.39502231420026124

To do: pull in additional metricsupdated heat map / correlation
   
   
Heat map: means = df_ml.groupby(level='CountryName').mean()
means = df_ml.mean()
corr = means.corr()
ax = sns.heatmap(corr)

#' VC.IDP.TOCV']  #‘Internally displaced persons, total displaced by conflict and violence (number of people)’
#‘Educational attainment, at least completed primary, population 25+ years, total (%) (cumulative)’, 
#IQ.CPA.TRAN.XQ ‘CPIA transparency, accountability, and corruption in the public sector rating (1=low to 6=high)’, 